fortran77/Fortran77Parser.g4

```diff
@@ -382,7 +382,7 @@ firstIfBlock
    ;
 
 elseIfStatement
-   : (ELSEIF | (ELSE IF)) LPAREN logicalExpression RPAREN THEN EOL? wholeStatement+
+   : (ELSEIF | (ELSE IF)) LPAREN logicalExpression RPAREN THEN EOL? commentStatement* (wholeStatement commentStatement*)+
```

Comment:
Does it make sense to put comments to the `HIDDEN` channel (`-> channel(HIDDEN)`)? 

https://github.com/antlr/grammars-v4/blob/38857b99fbb6effa129e2d49182d3814013f8e46/fortran77/Fortran77Lexer.g4#L652-L654

In this case there won't be need of using `commentStatement` on parser level.

Reply:
Sorry, but I don't think it's a good fix of the current problem. In the current implementation `commentStatement` is placed to certain parse rules but is not placed to others. That's why potential bugs still can be encountered. I suggest using standard technics for ignoring comments:

* Add `-> channel(HIDDEN)` to [`COMMENT`](https://github.com/antlr/grammars-v4/blob/master/fortran77/Fortran77Lexer.g4#L652) lexer rule. In this case lexer just puts tokens to another channel that is invisible for parser. `HIDDEN` channel is reserved in ANTLR and it's not neccessary to introduce a new one. Read more about lexer commands in the [ANTLR doc](https://github.com/antlr/antlr4/blob/master/doc/lexer-rules.md#channel).
* Remove `commentStatement` and its using from the parser.

Reply:
OK. I will try to update the grammar according to your suggestions. Thanks for pointing it out to me. 👍